HW5.ts

```diff
@@ -0,0 +1,81 @@
+interface Printable {
+  print(): void;
+}
+
+abstract class Shape {
+  readonly name: string;
+  readonly color: string;
+
+  constructor(name: string, color: string) {
+    this.name = name;
+    this.color = color;
+  }
+
+  abstract calculateArea(): number;
+}
+
+class Circle extends Shape {
+  radius: number;
+
+  constructor(name: string, color: string, radius: number) {
+    super(name, color);
+    this.radius = radius;
+  }
+
+  calculateArea(): number {
+    return Math.PI * Math.pow(this.radius, 2);
+  }
+}
+
+class Rectangle extends Shape implements Printable {
+  sideOne: number;
+  sideTwo: number;
+
+  constructor(name: string, color: string, sideOne: number, sideTwo: number) {
+    super(name, color);
+    this.sideOne = sideOne;
+    this.sideTwo = sideTwo;
+  }
+
+  calculateArea(): number {
+    return this.sideOne * this.sideTwo;
+  }
+
+  print(): void {
+    console.log(`Area of ${this.name} (${this.color}) is: length * width`);
+  }
+}
+
+class Square extends Shape {
```

Comment:
Looks like you missed something important - don't forget to force that class to implement the Printable interface. This will make sure everything runs smoothly and you don't encounter any unexpected issues down the line.